static/src/views/authentication.vue

```diff
@@ -103,34 +107,77 @@ export default {
   methods: {
     async moveToNextPage() {
       if (this.showLoginPage) {
-        console.log("submit phone number");
-
-        this.showLoginPage = this.show2FaPage = false;
-        this.showOtpPage = true;
+        try {
+          this.toastError = false;
+          const data = await this.$http.post("/login", {
+            phone: this.formattedPhoneNumber,
+          });
+          if (data && data.data && data.data.is_success) {
+            this.phoneCodeHash = data.data.phone_code_hash;
+            this.toastInput = "OTP is sent to registered number";
+            this.showLoginPage = this.show2FaPage = false;
+            this.showOtpPage = true;
+            this.showNotification();
+          }
+        } catch (e) {
+          this.toastError = this.nextButtonValidation = true;
+          this.toastInput = "Phone number is not registered";
+          this.showNotification();
+        }
       } else if (this.showOtpPage) {
-        this.showOtpPage = this.showOtpPage = false;
-        this.show2FaPage = true;
-        console.log("submit otp");
-        // const data = await this.$http.get(
-        //   "https://jsonplaceholder.typicode.com/todos/1"
-        // );
+        this.toastInput = "Verifing the OTP";
+        this.toastError = false;
+        this.showNotification();
+        try {
+          const data = await this.$http.post("/login", {
+            phone: this.formattedPhoneNumber,
+            code: this.otpNumber,
+            phone_code_hash: this.phoneCodeHash,
+          });
+          if (data && data.data && data.data.is_success) {
+            const { is_2FA_enabled = false } = data.data;
```

Comment:
I am not familiar with this syntax. What does this do?

Reply:
This is for enabling the 2FA page if the data is success. For line no. 137, it is a kind of technique used in Vue for cross-validation of data.